src/Operation/Plus.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace loophp\collection\Operation;
+
+use Closure;
+use Generator;
+use loophp\iterators\ConcatIterableAggregate;
+
+/**
+ * @immutable
+ *
+ * @template TKey
+ * @template T
+ */
+final class Plus extends AbstractOperation
+{
+    /**
+     * @template UKey
+     * @template U
+     *
+     * @return Closure(iterable<UKey, U>): Closure(iterable<TKey, T>): iterable<int|TKey|UKey, T|U>
+     */
+    public function __invoke(): Closure
+    {
+        $comparatorCallback =
+            /**
+             * @param T $left
+             *
+             * @return Closure(T): bool
+             */
+            static fn (mixed $left): Closure =>
+                /**
+                 * @param T $right
+                 */
+                static fn (mixed $right): bool => $left === $right;
+
+        return
+            /**
+             * @param iterable<UKey, U> $items
+             *
+             * @return Closure(iterable<TKey, T>): iterable<int|TKey|UKey, T|U>
+             */
+            static fn (iterable $items): Closure =>
+                /**
+                 * @param iterable<TKey, T> $iterable
+                 *
+                 * @return Generator<int|TKey|UKey, T|U>
+                 */
+                static fn (iterable $iterable): Generator => yield from (new Distinct())()($comparatorCallback)(static fn (mixed $value, mixed $key): mixed => $key)(new ConcatIterableAggregate([$iterable, $items]));
```

Comment:
@drupol , does it make any sense to remove duplicated indexes in the existing collection? I assume it would be more apparent if these were kept. Consider following test (fails with current implementation):

```php
public function testPlusDoesNotRemoveExistingDuplicatedIndices(): void
{
    $collection = Collection::fromIterable([
        ['id' => 1],
        ['id' => 2],
    ]);

    $result = $collection
        ->unwrap()
        ->plus(['id' => 3, 'id2' => 4])
        ->wrap()
        ->all(false)
    ;

    self::assertSame([
        ['id' => 1],
        ['id' => 2],
        ['id2' => 4],
    ], $result);
}
```

Moreover, it would be really pity if `plus([])` would drop all key duplicates. The following test fails as well:

```php
public function testPlusEmptyArrayDoesNotModifyCollection(): void
{
    $collection = Collection::fromIterable([
        ['id' => 1],
        ['id' => 2],
    ]);

    $result = $collection
        ->unwrap()
        ->plus([])
        ->wrap()
        ->all(false)
    ;

    self::assertSame([
        ['id' => 1],
        ['id' => 2],
    ], $result);
}
```

Reply:
Very nice catch and test case. I will see how we can improve the implementation. Feel free to come up with ideas.

Reply:
I updated the tests and the implementation, this is a bit uglier, but AFAIK there's no other alternative.